### DISCUSSION TITLE
Usage of System vs. LightningModule+LightningDataModule

### DISCUSSION THREAD

**Discussion by YairAmar** [asteroid-team/asteroid#654] [Ideas]
Looking at some simple examples of the usage of System in Asteroid, it seems not to have any benefit on the option of using 
LightningModule+LightningDataModule, while also not preserving the cool idea in Lightning where the datamodule and the model+optimizer+etc module are agnostic and one can be changed without changing the other.
In my opinion the System class should be removed.

**Comment by mpariente** [comment]
This is a very good point, thank you for your feedback. 

It was initially to keep the model, optimizer, data independant from the code of the LightningModule to make things clearer, and keep the users from having to write the training logic every time. But Lightning has evolved a LOT since then, and I agree that it doesn't make sense anymore. 

Do you want to contribute to move to the next/new version of Asteroid ?

**Reply by YairAmar** [reply]
I would love to!
How can join the effort?

**Reply by mpariente** [reply]
There is currently no main effort to do that, sadly. It requires quite a lot of bandwidth, which I don't have. But if you'd like to contribute, I'll review every suggestion and PR !
